### PR TITLE
Fix sentry errors from recording logics

### PR DESCRIPTION
**Pull request by rcmarron**
## Changes

2 small fixes that should resolve [this error](https://sentry.io/organizations/posthog2/issues/2778971871/?project=1899813&query=is%3Aunresolved) and [this one](https://sentry.io/organizations/posthog2/issues/2770942308/?project=1899813)

## How did you test this code?

For the first error, it would happen if the user 
1. opens a recording
2. closes the recording
3. resizes the window

For the first error, it would happen if the user 
1. opens a recording from the persons page
2. closes the recording using the back button

I manually verified both don't happen anymore

### CODE REVIEW

**Review by alexkim205 — APPROVED** [review]
It's a little strange how the action needs to be cached but lgtm if this fixes things!

**Comment by rcmarron** [comment]
> It's a little strange how the action needs to be cached but lgtm if this fixes things!

Yeah, it's because we need a stable reference to the event listener to allow JS to actually remove it. Looks like we use cache a few other places for this use case.

**Pull request merged**
